src/main/kotlin/demo/controllers/ApplicationController.kt

```diff
@@ -17,13 +18,42 @@ class ApplicationController(@Autowired private val applicationService: Applicati
 
     @CrossOrigin
     @PostMapping("/apply/{offerId}/user")
-    fun applyForAnOffer(@PathVariable offerId: Int, @RequestBody applicationPayload: ApplicationLoggedInPayload) =
-        applicationService.insertApplicationLoggedInUser(offerId, applicationPayload)
+    fun applyForAnOffer(@PathVariable offerId: Int, @RequestBody applicationPayload: ApplicationLoggedInPayload): ResponseEntity<Void> {
+        val saved = applicationService.insertApplicationLoggedInUser(offerId, applicationPayload)
+        return ResponseEntity.created(URI("/api/applications/" + saved.id)).build()
+    }
 
     @CrossOrigin
     @PostMapping("/apply/{offerId}/no-user")
-    fun applyForAnOffer(@PathVariable offerId: Int, @RequestBody applicationPayload: ApplicationNoUserPayload) =
-        applicationService.insertApplicationWithoutUser(offerId, applicationPayload)
+    fun applyForAnOffer(@PathVariable offerId: Int, @RequestBody applicationPayload: ApplicationNoUserPayload): ResponseEntity<Void> {
+        val saved = applicationService.insertApplicationWithoutUser(offerId, applicationPayload)
+        return ResponseEntity.created(URI("/api/applications/" + saved.id)).build()
+    }
+
+    @CrossOrigin
+    @GetMapping("/{applicationId}")
+    fun getApplicationById(@PathVariable applicationId: Int) =
+        applicationService.getApplicationById(applicationId)
+
+    @CrossOrigin
+    @GetMapping("/")
```

Comment:
Wydaje mi się, że tego "/" nie musi tu być

Reply:
Jakieś to samotne bez tej linijki mi się wydawało